src/utils/utils_concurrency.h

```diff
@@ -27,10 +29,17 @@
 #else /* __cplusplus */
 #include <atomic>
 #define _Atomic(X) std::atomic<X>
+
+using std::memory_order_acq_rel;
```

Comment:
Why do we need C++ code here?

Reply:
this code was indirectly included in c++ tests..
moved to test/base.hpp

Reply:
@bratpiorka but C++ code is still there and in the `test/base.hpp` only Copyright is changed

Reply:
Yes, I tried moving it to base.hpp, but that broke the building of examples. I also attempted to remove the __cplusplus code entirely, but it's not that simple. So, in fact, I'm not adding new C++ code here; I'm extending the existing code. Further refactoring should be done in a separate PR.

Reply:
> Yes, I tried moving it to base.hpp, but that broke the building of examples. 
Why? Our examples should not use any code either from umf internals, nor tests

>I also attempted to remove the __cplusplus code entirely, but it's not that simple. So, in fact, I'm not adding new C++ code here; I'm extending the existing code. Further refactoring should be done in a separate PR.
You are adding extra using, and stil we do not know why they are needed.